contrib/gitian-build.py

```diff
@@ -0,0 +1,242 @@
+    parser.add_argument('-S', '--setup', action='store_true', dest='setup', help='Set up the Gitian building environment. Only works on Debian-based systems (Ubuntu, Debian)')
+    parser.add_argument('-D', '--detach-sign', action='store_true', dest='detach_sign', help='Create the assert file for detached signing. Will not commit anything.')
+    parser.add_argument('-n', '--no-commit', action='store_false', dest='commit_files', help='Do not commit anything to git')
+    parser.add_argument('signer', help='GPG signer to sign each build assert file')
+    parser.add_argument('version', help='Version number, commit, or branch to build. If building a commit or branch, the -c option must be specified')
+
+    args = parser.parse_args()
+    workdir = os.getcwd()
+
+    args.linux = 'l' in args.os
+    args.windows = 'w' in args.os
+    args.macos = 'm' in args.os
+
+    args.is_bionic = b'bionic' in subprocess.check_output(['lsb_release', '-cs'])
+
+    if args.buildsign:
+        args.build = True
+        args.sign = True
+
+    args.sign_prog = 'true' if args.detach_sign else 'gpg --detach-sign'
+
+    args.lxc = (args.virtualization == 'lxc')
+    args.kvm = (args.virtualization == 'kvm')
+    args.docker = (args.virtualization == 'docker')
+
+    script_name = os.path.basename(sys.argv[0])
+    # Set all USE_* environment variables for gitian-builder: USE_LXC, USE_DOCKER and USE_VBOX
+    os.environ['USE_VBOX'] = ''
+    if args.lxc:
+        os.environ['USE_LXC'] = '1'
+        os.environ['USE_DOCKER'] = ''
+        if 'GITIAN_HOST_IP' not in os.environ.keys():
+            os.environ['GITIAN_HOST_IP'] = '10.0.3.1'
+        if 'LXC_GUEST_IP' not in os.environ.keys():
+            os.environ['LXC_GUEST_IP'] = '10.0.3.5'
+    elif args.kvm:
+        os.environ['USE_LXC'] = ''
+        os.environ['USE_DOCKER'] = ''
+    elif args.docker:
+        os.environ['USE_LXC'] = ''
+        os.environ['USE_DOCKER'] = '1'
+    else:
+        print(script_name+': Wrong virtualization option.')
+        print('Try '+script_name+' --help for more information')
+        exit(1)
+
+    # Signer and version shouldn't be empty
+    if args.signer == '':
+        print(script_name+': Missing signer.')
+        print('Try '+script_name+' --help for more information')
+        exit(1)
+    if args.version == '':
+        print(script_name+': Missing version.')
+        print('Try '+script_name+' --help for more information')
+        exit(1)
+
+    # Add leading 'v' for tags
+    if args.commit and args.pull:
+        raise Exception('Cannot have both commit and pull')
+    args.commit = ('' if args.commit else 'v') + args.version
+
+    if args.setup:
+        setup()
+
+    if not args.build and not args.sign and not args.verify:
+        exit(0)
+
+    # Disable for MacOS if no SDK found
+    if args.macos and not os.path.isfile('gitian-builder/inputs/MacOSX10.11.sdk.tar.gz'):
```

Comment:
We implemented automatic downloading of the MacOS sdk as part of gitian-build.sh. Maybe we should do the same here as well?

Reply:
See https://github.com/dashpay/dash/pull/2178/commits/65a04afea0d655a0ff7fc284322f6ccf6f6728a7

Reply:
Good point. Backports shouldn't drop already implemented features (unless that's the point of the original PR we are backporting).